### ISSUE TITLE
Can not subscribe any message from topic "/ridgeback_velocity_controller/odom"

### ISSUE THREAD

**Issue by WshgL** [ridgeback/ridgeback#27]
I get some problem with odom and tf:

(1) I can not subscribe any message from topic "/ridgeback_velocity_controller/odom", But the "/status" topic is working properly. I can get the output voltage information of the motor driver from the "/status" topic. In addition, I can also control robot movement through the "/cmd_vel" topic.

(2) Because no message is available from the topic "/ridgeback_velocity_controller/odom", "/odometry/filtered" topic is also problematic. "/odometry/filtered" topic has twist information, its pose information is constantly increasing, even though my robot is still. Eventually the transformation of odom to base_link is also wrong.
[screen_shot_1](https://p20.zdusercontent.com/attachment/126981/AcDR6WrvGNSJ95SVuRkhQ5QUZ?token=eyJhbGciOiJkaXIiLCJlbmMiOiJBMTI4Q0JDLUhTMjU2In0..7-pVFeRgxwO1AKFc6w6IXg.tAoan8Des82WWklEYuhm_RChsXef20e8W3y5MdACKULaB4yCoLJZDQ7f7q2QqrP_1vYCj_gB6hbcx6gNAotYjfDPlwaVGMrNRLMmMSWWZHcs5IsWxQ-dY2XqSSNbqU1F99ltpDR0Ofh_2LcHzJYgou3NmBjb5X8T-rd6OX3YKkHKngAtJw2HxI-r8V_JOG62UCqxLiJ0eIfF6oHYoGRDtf8RFqy6qP_zQ-_Et-oAmIiER5ZqG9Xhuwkdo-BmahWJRIOlIwDlG3QjyGkU_ZwobFzwfqABRYHb9qibk5IxskQ.55m77t9Ck48lv0dn2lIYCw)

(3) When I executed "roslaunch ridgeback_viz view_robot.launch", the rviz interface told me that: the tf information is incomplete and there are multiple links that are not connected to the odom. In addition, there are some problems with time in the tf transformation.

```bash
No transform from [front_left_wheel_link] to [odom] 
No transform from [front_right_wheel_link] to [odom] 
No transform from [front_rocker_link] to [odom] 
No transform from [rear_left_wheel_link] to [odom] 
No transform from [rear_right_wheel_link] to [odom]

For frame [front_laser]: No transform to fixed frame [odom]. TF error: [Lookup would require extrapolation into the future. Requested time 1545108932.845590104 but the latest data is at time 36280661.233775787, when looking up transform from frame [front_laser] to frame [odom]]

Message removed because it is too old (frame=[imu_link], stamp=[34.113000000])
```
[screen_shot_2](https://p20.zdusercontent.com/attachment/126981/HUoW8lgb2JGLL6Jpy9CGpk3Cc?token=eyJhbGciOiJkaXIiLCJlbmMiOiJBMTI4Q0JDLUhTMjU2In0..PPsHllSHPivArVDOtRqwvA.Msilr4vBQhsHXwbv6bEiHySiVcGQvTmDHQmJv9xm-KRLtBH0bvMYwx9sC9uMIbEzaThpkdFAPv7YKfZYmpy9ehzt3oCBzwhZS8QF7pmT4mih3dI8crH0dN8chzQxql1S-XjRFzf3Z7RpbpY_LjgMoGjaLPGlKKcLbHE9wQWL9L924QKYa9ZY6XSCy2uivE2FbUEqhe2rqr8uav79oCv1ReKX3AlPMwJefUwDld6O_U0MRXzHrKZJz9-AMWkReY_Ma9fnc7F0XjDplS9DfXyksfnYAyuzCcDDf7F2S8VBXXs.7yGImNus6DOUz9tVeygKfQ)
[screen_shot_3](https://p20.zdusercontent.com/attachment/126981/GEEEnVFmiem8uw8GFtRXeuvV0?token=eyJhbGciOiJkaXIiLCJlbmMiOiJBMTI4Q0JDLUhTMjU2In0..7bBie5kYz0PIwMyRdn5jQw.qi_yjiF6-hYoVThw7Xr6YBe4sc4MAsMjiwMu9fSWkoZ1cpe82NRlGkN02gqPrV62ygIWpoLUwpsZx-fJa-io8jzWzbWkDWX9ioeYV95DYae-50Pd3_dJydzUiertju4aCIawyevG6qK1FzO0ffHcHGVTmA7cFiwhvjpkIShq5vjnovACBAn7aw8CR2oo52x8gV5aUyVeFZ2SeK7u8Fn7VncgdLdUnNQkcaKMyRFJhhRPeL_FYaGz40laRvHcKwlzAIXS6ImGpskO8XkkYzjrHuB57TOumC0hTsermAPIGL4.byloo9GkhY47ytMtk6hfUg)
[screen_shot_4](https://p20.zdusercontent.com/attachment/126981/u231xgoyLfULpYx2LXeJmcZBn?token=eyJhbGciOiJkaXIiLCJlbmMiOiJBMTI4Q0JDLUhTMjU2In0..8LdLxc4-322YJpf09cxHGw.oXx1J3konwMtCY4pXw9DkVQ264ElCSi5C6DFRCY5YUGnQ4-zA0k899BKVv2H3n7BxHTD4Fode1XrVcwP_cvTT4RQ7FJk_9-N8r2iOdijwrtvzzyoJ4V0SUaCxjhR56RLtNEEfx5r4AuN3a7Rqsi7qfsnqQv_DlDT2FcBwmDQV7aXNXbyOuUBYLLUsI0JIyYi_D5qyWauE9SqjA_HzeoTMkFxGbX9ISY5iv_uaDjKtkPdLZ0Mm1fMY9alfEMuPqS9FuuJOFJpluf--IyMark89vSZwMVFSHbmn2dnC_tMtNs.YVQYcEVrtgFAXrXJZ8DZ-w)

**Comment by tonybaltovski** [comment]
Are you getting any errors when this is starting up?  For Indigo, you could do `sudo tail -n 100 /var/log/upstart/ros.log`.  For Kinetic, `sudo journalctl -fu ros`.

**Comment by WshgL** [comment]
> Are you getting any errors when this is starting up? For Indigo, you could do `sudo tail -n 100 /var/log/upstart/ros.log`. For Kinetic, `sudo journalctl -fu ros`.

Thanks for your reply,

I get the flowing message from `sudo journalctl -fu ros`
```bash
12月 18 21:32:07 ridgeback ros-start[1035]: Warning: TF_OLD_DATA ignoring data from the past for frame front_rocker_link at time 2.48303e+09 according to authority unknown_publisher
12月 18 21:32:07 ridgeback ros-start[1035]: Possible reasons are listed at http://wiki.ros.org/tf/Errors%20explained
12月 18 21:32:07 ridgeback ros-start[1035]:          at line 277 in /tmp/binarydeb/ros-kinetic-tf2-0.5.20/src/buffer_core.cpp
```

I want to provide more information:

I downloaded the latest system installation image from [here](https://packages.clearpathrobotics.com/stable/images/). And then install it into my Ridgeback.
After install the Ubuntu 16.04, I am troubled by a problem: 
Sometimes ridgeback can start properly, but sometimes, its indicator light will remain "solid red" and then automatically power off quickly. About 50% of the two cases.
But because at least there is a time to successfully start the robot, I temporarily ignored this problem.

Then the another problem is this issue.

Are these two problems related to each other?

Look forward to your reply, thank you very much!
Best regards

**Comment by wilsonn-leong** [comment]
> Are you getting any errors when this is starting up? For Indigo, you could do `sudo tail -n 100 /var/log/upstart/ros.log`. For Kinetic, `sudo journalctl -fu ros`.

When I power on the Ridgeback, I get some error message:
```bash
"Failed to start LSB: IPv4 DHCP client with IPv4LL support" 
"Failed to start Raise network interfaces" 
"Failed to start Bluetooth service" 
"Dependency failed for ds4drv daemon"
```

Please refers to

[photo_1](https://p20.zdusercontent.com/attachment/126981/rZpw7mBcwjC193Q30pNWVaZhL?token=eyJhbGciOiJkaXIiLCJlbmMiOiJBMTI4Q0JDLUhTMjU2In0..n71Tvmux2_upQkcrKvzP2Q.DllhlN9cCYYe9t_KIrofss75ceh_klwhJb0eS3RkWouuFs3rlx6CK1M9zAitHxXUMv3-XBAuaDbAZmmCQaxgxMiN9zdeF3xm9Gc-XdCxLSDHBTNN4vmZmVNzt7YXwdipXepCy1mOTk7db_q58pnM8p2QV3OpSAiHALzfX-BttqyptP0vdOQ7PYTK2DO1rgHScxOwHaJLQXtdVDnGcgZK1MkTKX6vGqMGfrXU_3kSfYaRkky7y97DOSZh0J2rNicZMVDyRIZzX-E_N6pyx9xmaBIrC0KpewK5dlG-gXZNuEc.OlF4cT5xbSMI-MDUQ0mTqg)
[photo_2](https://p20.zdusercontent.com/attachment/126981/wCBrkYi5dKuwduWL4UOKLysjf?token=eyJhbGciOiJkaXIiLCJlbmMiOiJBMTI4Q0JDLUhTMjU2In0..giXl7lP8FulqLJVYRo21kg.yk4bMj5DmOEZZws1F9PWTV3b52vUid55FBa9wPisoxfSmmTaXk6QbVnRM-YM8pwOIesHxiQbp0mgwTyzzhJr2dR7ci48uyLbYcAKN_54NW_tmI8BdnIIFGF-m6wkp_Mt3yzEXTml24BvPc-vudxhF1DiujTHxQjjwnGIR4YQf8bNjnBmbO2WqLSoK9nrz0_5x5Uqqx7nmprkOzfwgmzRf0J2CAU_yoV_CGaYZplPlV_5NQm2Ptmcv5tSt5BId1jBQ1ichHbZq5K6XWLU9yFU4niFxCCo2ptHcwg0jehxuQQ.mbR_n0pGxwiG9nTxkxafLw)
[photo_3](https://p20.zdusercontent.com/attachment/126981/DoKHKM2CCwiwg06wbIiZI0qBb?token=eyJhbGciOiJkaXIiLCJlbmMiOiJBMTI4Q0JDLUhTMjU2In0..iTteC0kT1Rz4b4oIHkapsw.NS9mZWwikTlcTG_XRcC7TCJj8UjgAt_OR4WAQndvHnfSKw0KSE6H4Yoxrq9jjJFHk2nYtKE6NpCedoFqdNJUWHNboHTpwcoG6b47ET5p0kwKjM_rRIkT-RJWBVQNbKQrV3DDRHISNlO-jhHSVv-PkbJTjyIgjxa_my5zFCs4Wy80xfSmoKNIEckiA_RXAgYd3T1vYEwyuAR4XoXMk3ijLKQXqcyi8rQms79--0y2a3eYTK_xmYtrVluccP4Ya2upi5e16yFF8y1VjBgVyounYBltHLq2Ttax7-cW0jdZIRY.EHRtKZf-r4j7uNPwgCYb2w)
[photo_4](https://p20.zdusercontent.com/attachment/126981/gYYeZgAKo9pY1FW8A95xnNH5d?token=eyJhbGciOiJkaXIiLCJlbmMiOiJBMTI4Q0JDLUhTMjU2In0..WBi8DdB5XWauUYjYHqiiwA.84z946UyWbeaClwakyrs4smi4CY875dkC5AJTNVuvHx3r33dTzt7aG3h9JzGEJcKQr6HLCvSSlM8wX-owJroprnvF6R8guCe3Ls222wocK2WSDFcr_-f3UtFRWl1TsThYQvJNXdg4ZOyFQ3cRGnJyL6TDmyyfQeSGOaGq6q90BM3QPge8wiU9jAgF7ylkHG4DxmapRP9Kf_iza72vkYc_3Ma17YH5B8hul2aE3QEoFlFi-3AJ4709kXr-eJ7LxyHeQ6vb0iMXCQFL3tkABSFJBYLa2bGVWIey-0uYQeDpk8.f9VTISEJSSxuHCYeimDpyQ)

**Comment by wilsonn-leong** [comment]
> Are you getting any errors when this is starting up? For Indigo, you could do `sudo tail -n 100 /var/log/upstart/ros.log`. For Kinetic, `sudo journalctl -fu ros`.

@tonybaltovski 

I want to provide more information. 
I run rqt_console and get some imformation

The case of "solid red" indicator light :
[shut_down_case](https://user-images.githubusercontent.com/43674763/50202923-bea76880-039a-11e9-9b94-ed8880d7098d.png)

The case of start properly  :
[normal_case_1](https://user-images.githubusercontent.com/43674763/50202924-bea76880-039a-11e9-909f-b9a362401958.png)
[normal_case_2](https://user-images.githubusercontent.com/43674763/50202925-bea76880-039a-11e9-901a-2d097eef3356.png)

**Comment by wilsonn-leong** [comment]
> Are you getting any errors when this is starting up? For Indigo, you could do `sudo tail -n 100 /var/log/upstart/ros.log`. For Kinetic, `sudo journalctl -fu ros`.

@tonybaltovski 

I found the reason that Ridgeback will keep the "solid red" indicator and automatically power off.
This is because `ridgeback_node` has died.

**Comment by WshgL** [comment]
> Are you getting any errors when this is starting up? For Indigo, you could do `sudo tail -n 100 /var/log/upstart/ros.log`. For Kinetic, `sudo journalctl -fu ros`.

I have solved this problems by replacing the ros package in the clearpath repository with the latest ros package on github.

But I am troubled by the another problem with WIFI connection.
Please refer to [issue #24 in ridgeback_robot
](https://github.com/ridgeback/ridgeback_robot/issues/24#issue-392612927)

**Comment by tonybaltovski** [comment]
@wilsonn-leong when did you install the image?  I believe this is most likely an ABI issue with ros-controls.

**Comment by WshgL** [comment]
@tonybaltovski 
I download the image from [this repertory](https://packages.clearpathrobotics.com/stable/images/latest/), the latest version 0.3.4. I installed this image a week ago. The ros package in the clearpath repository is automatically downloaded and installed during the installation process. I have solved this problems by replacing the ros package in the clearpath repository with the latest ros package on github.
